modules/prebidServerBidAdapter/index.js

```diff
@@ -395,13 +397,97 @@ const OPEN_RTB_PROTOCOL = {
           return { w, h };
         });
 
-        banner = {format};
+        mediaTypes['banner'] = {format};
       }
 
-      let video;
       const videoParams = utils.deepAccess(adUnit, 'mediaTypes.video');
       if (!utils.isEmpty(videoParams)) {
-        video = videoParams;
+        mediaTypes['video'] = videoParams;
+      }
+
+      const nativeParams = processNativeAdUnitParams(utils.deepAccess(adUnit, 'mediaTypes.native'));
+      if (nativeParams) {
+        let assetCounter = 200;
+        try {
+          mediaTypes['native'] = {
+            request: JSON.stringify({
+              // TODO: determine if following OpenRTB recommended properties can be defined or are needed
+              // context: int,
+              // plcmttype: int,
+              // privacy: int
+              assets: Object.keys(nativeParams).reduce((assets, type, index) => {
+                let params = nativeParams[type];
+
+                function newAsset(obj) {
+                  return Object.assign({
+                    id: assetCounter++,
+                    required: params.required ? 1 : 0
+                  }, obj ? utils.cleanObj(obj) : {});
+                }
+
+                if (
+                  (type === 'image' || type === 'icon')
+                ) {
+                  let imgTypeId = ({
+                    'icon': 1,
+                    'image': 3
+                  })[type];
+
+                  if (Array.isArray(params.sizes)) {
+                    let sizes = params.sizes;
+                    if (!Array.isArray(sizes[0])) { // support single size or an array of sizes
+                      sizes = [sizes]
+                    }
+                    // TODO: Currently creating multiple image assets for each size.  Figure out if this is correct!
+                    sizes.forEach(size => {
+                      assets.push(newAsset({
+                        img: {
+                          type: imgTypeId,
+                          w: size[0],
+                          h: size[1]
+                        }
+                      }));
+                    });
+                  // TODO: Figure out what to do about aspect_ratios, ext perhaps?
+                  // } else if (Array.isArray(params.aspect_ratios)) {
+                  } else {
+                    assets.push(newAsset({
+                      img: {
+                        type: imgTypeId
+                      }
+                    }));
+                  }
+                } else if (type === 'title') {
+                  assets.push(newAsset({
+                    title: {
+                      len: params.len
+                    }
+                  }));
+                } else {
+                  let dataAssetTypeId = ({
+                    sponsoredBy: 1,
+                    body: 2,
+                    cta: 12,
+                    // TODO: clickUrl in request object?
```

Comment:
I'm not sure how or if `clickUrl` maps to any of the data types defined in the OpenRTB Native 1.2 spec.  I'm assuming that it does not map to anything in the request object and is assumed required for all bid responses (since ads aren't much good if you can't click on them to go somewhere).